src/hooks/useIntersection.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react';
+const useIntersection = (
+    loadingRef: React.MutableRefObject<HTMLDivElement | null>,
+    bottomBoundaryRef: React.MutableRefObject<HTMLDivElement | null>,
+    onLoadMore: () => void
+) => {
+    const [isIntersecting, setIsIntersecting] = useState(false);
+
+    useEffect(() => {
+        const options = {
+            root: null,
+            rootMargin: '0px',
+            threshold: 0,
+        };
+
+        const observer = new IntersectionObserver(([entry]) => {
+            setIsIntersecting(entry.isIntersecting);
+        }, options);
+
+        if (loadingRef.current) {
+            observer.observe(loadingRef.current);
+        }
+
+        return () => {
+            if (loadingRef.current) {
+                observer.unobserve(loadingRef.current);
+            }
+        };
+    }, [loadingRef]);
+
+    useEffect(() => {
+        if (isIntersecting) {
+            onLoadMore();
+        }
+    }, [isIntersecting, onLoadMore]);
+
+    useEffect(() => {
```

Comment:
Why is the scroll event also required for this hook? IntersectionObsever alone should suffice.

Reply:
IntersectionObserver has some limitations, and using both approaches ensures better handling of the load more functionality.